core/iwasm/interpreter/wasm_loader.c

```diff
@@ -4425,6 +4442,11 @@ wasm_loader_unload(WASMModule *module)
     }
 #endif
 
+#if WASM_ENABLE_SHARED_MEMORY != 0
+    WASMModule *wasm_module = (WASMModule *)module;
+    os_mutex_destroy(&wasm_module->mem_lock);
```

Comment:
Had better not declare variable here, here we can just `os_mutex_destroy(&module->mem_lock);`

Reply:
I'll update it depending on the outcome of https://github.com/bytecodealliance/wasm-micro-runtime/pull/1960#discussion_r1106963947